File: examples/flexbox-gap.rs
use taffy::prelude::*;

// Creates three 20px x 20px children, evenly spaced 10px apart from each other
// Thus the container is 80px x 20px.

fn main() -> Result<(), taffy::error::TaffyError> {
    let mut taffy = Taffy::new();

    let child_style = Style { size: Size { width: points(20.0), height: points(20.0) }, ..Default::default() };
    let child0 = taffy.new_leaf(child_style.clone())?;
    let child1 = taffy.new_leaf(child_style.clone())?;
    let child2 = taffy.new_leaf(child_style.clone())?;

    let root = taffy.new_with_children(
        Style { gap: Size { width: points(10.0), height: zero() }, ..Default::default() },
        &[child0, child1, child2],
    )?;

    // Compute layout and print result
    taffy.compute_layout(root, Size::MAX_CONTENT)?;
    taffy::debug::print_tree(&taffy, root);

    Ok(())
}

File: bindings/java/java/src/examples/java/com/dioxuslabs/taffy/examples/FlexboxGap.java
package com.dioxuslabs.taffy.examples;

import com.dioxuslabs.taffy.AvailableSpace;
import com.dioxuslabs.taffy.Dimension;
import com.dioxuslabs.taffy.LengthPercentage;
import com.dioxuslabs.taffy.Style;
import com.dioxuslabs.taffy.TaffyTree;

/**
 * Java port of {@code examples/flexbox_gap.rs}.
 *
 * <p>Creates three 20x20 children, evenly spaced 10px apart. The container
 * therefore computes as 80x20.
 */
public final class FlexboxGap {

    private FlexboxGap() {}

    public static void main(String[] args) {
        try (TaffyTree tree = new TaffyTree()) {
            Style childStyle = Style.builder()
                    .size(Dimension.length(20f), Dimension.length(20f));

            long child0 = tree.newLeaf(childStyle);
            long child1 = tree.newLeaf(childStyle);
            long child2 = tree.newLeaf(childStyle);

            long root = tree.newWithChildren(Style.builder()
                            .gap(LengthPercentage.length(10f), LengthPercentage.ZERO),
                    child0, child1, child2);

            tree.computeLayout(root, AvailableSpace.MAX_CONTENT, AvailableSpace.MAX_CONTENT);

            System.out.println("root:   " + tree.layout(root));
            System.out.println("child0: " + tree.layout(child0));
            System.out.println("child1: " + tree.layout(child1));
            System.out.println("child2: " + tree.layout(child2));
        }
    }
}
